clang/lib/Format/UnwrappedLineParser.cpp

```diff
@@ -3914,7 +3914,15 @@ void UnwrappedLineParser::parseRecord(bool ParseAsExpr) {
   // (this would still leave us with an ambiguity between template function
   // and class declarations).
   if (FormatTok->isOneOf(tok::colon, tok::less)) {
```

Comment:
Why can't we use `parseAngle`? (Don't know if that would lead to better results.)

This can easily be broken with `class A<X > 3> f() {}`, right?

Reply:
Yes, that would be easily broken with this case.
I don't think it is possible to use `parseAngle` now, because it is defined in inside `AnnotatingParser` in `Format/TokenAnnotator.cpp` and it uses some state of the class.

Reply:
I've updated the patch because I forgot about the case of template specializations.
Template specializations should be OK now.